smart_open/gcs.py

```diff
@@ -127,475 +65,47 @@ def open(
         The name of the blob within the bucket.
     mode: str
         The mode for opening the object.  Must be either "rb" or "wb".
-    buffer_size: int, optional
-        The buffer size to use when performing I/O. For reading only.
     min_part_size: int, optional
         The minimum part size for multipart uploads.  For writing only.
     client: google.cloud.storage.Client, optional
         The GCS client to use when working with google-cloud-storage.
     blob_properties: dict, optional
         Set properties on blob before writing.  For writing only.
-
+    blob_open_kwargs: dict, optional
+        Set properties for opening the blob, passed through directly to
+        the google-cloud-storage library
     """
-    if mode == constants.READ_BINARY:
-        fileobj = Reader(
-            bucket_id,
+    if blob_open_kwargs is None:
+        blob_open_kwargs = {}
+    if blob_properties is None:
+        blob_properties = {}
+
+    if client is None:
+        client = google.cloud.storage.Client()
+
+    bucket = client.bucket(bucket_id)
```

Comment:
I could believe that for performance reasons we may not want to make this call here every time when getting an object. However when writing an object the error which is thrown if the bucket doesn’t exist is very obtuse and somewhat hard to decipher!